### PR TITLE
Pull in scaffolder-relation-processor plugin from @backstage-community to 1.2.x branch of janus-showcase because it is the new scope.

### PR DESCRIPTION
**Pull request by Fortune-Ndlovu**
This PR pulls in the scaffolder-relation-processor plugin from the [backstage-community repository](https://github.com/backstage/community-plugins/blob/main/workspaces/scaffolder-relation-processor/plugins/catalog-backend-module-scaffolder-relation-processor/package.json) into the [1.2.x branch of janus-showcase](https://github.com/janus-idp/backstage-showcase/blob/1.2.x/dynamic-plugins/imports/package.json). This includes updating the dynamic-plugins-imports/package.json file to reference the correct version of the plugin.

## Changes Made

Updated dynamic-plugins-imports/package.json to include the following dependency because of its newly allocated scope.
`"@backstage-community/plugin-catalog-backend-module-scaffolder-relation-processor": "0.2.0"`

## Which issue(s) does this PR fix

The [scaffolder-relation-processor plugin](https://github.com/backstage/community-plugins/tree/main/workspaces/scaffolder-relation-processor/plugins/catalog-backend-module-scaffolder-relation-processor) is required for enhancing the scaffolding capabilities of the catalog backend module. This update ensures that the janus-showcase project is using the latest version of this plugin from the backstage-community repository in the new scope defined.

## PR acceptance criteria

Please make sure that the following steps are complete:

- [x] Ensured that the updated package.json installs the correct version of the plugin.
- [x] GitHub Actions are completed and successful
- [x] Unit Tests are updated and passing
- [ ] E2E Tests are updated and passing
- [ ] Documentation is updated if necessary (requirement for new features)
- [ ] Add a screenshot if the change is UX/UI related

### CODE REVIEW

**Comment by Fortune-Ndlovu** [comment]
> @Fortune-Ndlovu: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:
> 
> Test name	Commit	Details	Required	Rerun command
> ci/prow/images	[9055751](https://github.com/janus-idp/backstage-showcase/commit/90557518ee1d28e256f53590fd400a3290308caa)	[link](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/janus-idp_backstage-showcase/1367/pull-ci-janus-idp-backstage-showcase-1.2.x-images/1807783439676477440)	true	`/test images`
> [Full PR test history](https://prow.ci.openshift.org/pr-history?org=janus-idp&repo=backstage-showcase&pr=1367). [Your PR dashboard](https://prow.ci.openshift.org/pr?query=is:pr+state:open+author:Fortune-Ndlovu).
> 
> Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md). If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).

/retest-required

**Comment by Fortune-Ndlovu** [comment]
> `/test e2e-tests`

/test e2e-tests

**Comment by Fortune-Ndlovu** [comment]
> /test e2e-tests

**Comment by nickboldt** [comment]
I'm confused. 

The new https://github.com/backstage/community-plugins/blob/main/workspaces/scaffolder-relation-processor/package.json#L3 is at version 1.0.0 and is marked private. 

But we have https://www.npmjs.com/package/@backstage-community/plugin-catalog-backend-module-scaffolder-relation-processor 0.2.0 from 21 days ago too. 

@Zaperex how is the 1.0.0 package.json related to the 0.2.0 release?

**Comment by Zaperex** [comment]
> I'm confused.
>
> The new https://github.com/backstage/community-plugins/blob/main/workspaces/scaffolder-relation-processor/package.json#L3 is at version 1.0.0 and is marked private.
> But we have https://www.npmjs.com/package/@backstage-community/plugin-catalog-backend-module-scaffolder-relation-processor 0.2.0 from 21 days ago too.

Ah so the `package.json` you referenced is the `package.json` of the workspace, not the package. The `package.json` of the package is [here](https://github.com/backstage/community-plugins/blob/main/workspaces/scaffolder-relation-processor/plugins/catalog-backend-module-scaffolder-relation-processor/package.json).

> Also there's a 1.1.1 version of https://www.npmjs.com/package/@janus-idp/backstage-plugin-catalog-backend-module-scaffolder-relation-processor-dynamic available from 20 days ago, which is presumably newer than the 1.0.3 or 0.2.0 versions in this PR.

Technically it's newer just because the `@janus-idp/cli` dev dependency is slightly newer.

**Comment by nickboldt** [comment]
OK so I'm hearing that https://github.com/backstage/community-plugins/blob/main/workspaces/scaffolder-relation-processor/plugins/catalog-backend-module-scaffolder-relation-processor/package.json#L4 should be bumped to 1.1.2 so it's larger than the 1.1.1 version out there, to eliminate the confusion around "0.2 is newer than 1.1" 

Would you agree with that ?

**Comment by Zaperex** [comment]
one main concern would just be semvar being broken, but when did backstage ever follow it :joy:

**Comment by nickboldt** [comment]
Blocked by some migration work to be done in https://issues.redhat.com/browse/RHIDP-3162

**Review by nickboldt — CHANGES_REQUESTED** [review]
See https://issues.redhat.com/browse/RHIDP-3162

File: dynamic-plugins/imports/package.json [hunk 1]
```diff
@@ -34,6 +34,6 @@
     "@janus-idp/backstage-scaffolder-backend-module-regex": "1.4.12",
     "@janus-idp/backstage-scaffolder-backend-module-servicenow": "1.4.14",
     "@janus-idp/backstage-scaffolder-backend-module-sonarqube": "1.4.12",
-    "@janus-idp/backstage-plugin-catalog-backend-module-scaffolder-relation-processor": "1.0.3"
+    "@backstage-community/backstage-plugin-catalog-backend-module-scaffolder-relation-processor": "1.2.1"
```
Comment (nickboldt) [inline dynamic-plugins/imports/package.json:37]: can you have a peer dependency from another repo?

**Comment by nickboldt** [comment]
docker build failed with 

```
#28 79.89 Error: Command failed: npm pack @backstage-community/backstage-plugin-catalog-backend-module-scaffolder-relation-processor@1.2.1
#28 79.89     at genericNodeError (node:internal/errors:984:15)
#28 79.89     at wrappedFn (node:internal/errors:538:14)
#28 79.89     at checkExecSyncError (node:child_process:890:11)
#28 79.89     at execSync (node:child_process:962:15)
#28 79.89     at Object.<anonymous> (/opt/app-root/src/dynamic-plugins/imports/import-plugins.js:35:19)
#28 79.89     at Module._compile (node:internal/modules/cjs/loader:1369:14)
#28 79.89     at Module._extensions..js (node:internal/modules/cjs/loader:1427:10)
#28 79.89     at Module.load (node:internal/modules/cjs/loader:1206:32)
#28 79.89     at Module._load (node:internal/modules/cjs/loader:1022:12)
#28 79.89     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:135:12) {
#28 79.89   status: 1,
#28 79.89   signal: null,
#28 79.89   output: [ null, Buffer(0) [Uint8Array] [], null ],
#28 79.89   pid: 730,
#28 79.89   stdout: Buffer(0) [Uint8Array] [],
#28 79.89   stderr: null
#28 79.89 }
#28 79.89 
#28 79.89 Node.js v20.12.2
#28 79.91 error Command failed with exit code 1.
#28 79.91 info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
#28 ERROR: process "/bin/sh -c now=$(date -u +%FT%TZ); sed -i packages/app/src/build-metadata.json -r     -e \"s/(\\\"Last Commit: .+\\\")/\\1, \\\"Build Time: $now\\\"/\" &&   cat packages/app/src/build-metadata.json; echo &&   $YARN build --filter=backend &&   $YARN --cwd ./dynamic-plugins/imports export-dynamic --no-install &&   $YARN --cwd ./dynamic-plugins/imports install-dynamic &&   $YARN export-dynamic -- --filter=./dynamic-plugins/wrappers/* &&   $YARN copy-dynamic-plugins dist" did not complete successfully: exit code: 1
------
```

File: dynamic-plugins/imports/package.json (hunk 1 above)
Comment (Fortune-Ndlovu) [reply dynamic-plugins/imports/package.json:37]: I think we can have a peer dependency from another repo to specify that our package relies on a specific version of another package

**Review by nickboldt — APPROVED** [review]
LGTM, I assume you have a similar fix for main branch in mind?

**Pull request merged**